proxy/envoy/config.go

```diff
@@ -183,13 +156,29 @@ func buildClusters(env proxy.Environment, role proxy.Node) (clusters Clusters) {
 func buildSidecar(env proxy.Environment, sidecar proxy.Node) (Listeners, Clusters) {
 	instances := env.HostInstances(map[string]bool{sidecar.IPAddress: true})
 	services := env.Services()
+	managementPorts := env.ManagementPorts(sidecar.IPAddress)
 
 	inbound, inClusters := buildInboundListeners(env.Mesh, sidecar, instances)
 	outbound, outClusters := buildOutboundListeners(env.Mesh, sidecar, instances, services, env)
+	mgmtListeners, mgmtClusters := buildMgmtPortListeners(env.Mesh, sidecar, managementPorts, sidecar.IPAddress)
 
 	listeners := append(inbound, outbound...)
 	clusters := append(inClusters, outClusters...)
 
+	// If management listener port and service port are same, bad things happen
+	// when running in kubernetes, as the probes stop responding. So, append
+	// non overlapping listeners only.
+	for _, m := range mgmtListeners {
+		l := listeners.GetByAddress(m.Address)
+		if l != nil {
+			glog.Warningf("Omitting listener for management address %s (%s) due to collision with service listener %s (%s)",
+				m.Name, m.Address, l.Name, l.Address)
+			continue
+		}
+		listeners = append(listeners, m)
+	}
+	clusters = append(clusters, mgmtClusters...)
```

Comment:
we shouldn't be adding clusters for non-existent listeners, they might override clusters for non-management listeners